__tests__/TestFuture.re

```diff
@@ -311,4 +311,38 @@ describe("Future Belt.Result", () => {
         |> finish
       );
   });
-});
+
+  testAsync("value recursion is stack safe", finish => {
```

Comment:
Make a deeply nested chain and test that we don't blow the stack there. Something like the following:
```reason
next(x)
->Future.flatMap(x => Future.value(x + 1))
->Future.map(x => x + 1)
->Future.flatMap(x => Future.value(x + 1)->Future.flatMap(x => Future.value(x + 1)))
```

Reply:
Done